fvm/transactionVerifier.go

```diff
@@ -14,6 +16,123 @@ import (
 	"github.com/onflow/flow-go/module/trace"
 )
 
+type signatureType struct {
+	message []byte
+
+	errorBuilder func(flow.TransactionSignature, error) errors.CodedError
+
+	aggregateWeights map[flow.Address]int
+}
+
+type signatureEntry struct {
+	flow.TransactionSignature
+
+	signatureType
+
+	accountKey flow.AccountPublicKey
+
+	verifyErr errors.CodedError
+}
+
+func (entry *signatureEntry) newError(err error) errors.CodedError {
+	return entry.errorBuilder(entry.TransactionSignature, err)
+}
+
+func (entry *signatureEntry) matches(
+	proposalKey flow.ProposalKey,
+) bool {
+	return entry.Address == proposalKey.Address &&
+		entry.KeyIndex == proposalKey.KeyIndex
+}
+
+func (entry *signatureEntry) verify() errors.CodedError {
+	valid, err := crypto.VerifySignatureFromTransaction(
+		entry.Signature,
+		entry.message,
+		entry.accountKey.PublicKey,
+		entry.accountKey.HashAlgo,
+	)
+	if err != nil {
+		entry.verifyErr = entry.newError(err)
```

Comment:
this would be an exception, better to add some context: 
```
entry.verifyErr = entry.newError(fmt.Errorf("failed to verify signature: %w", err))
```